ten/one_david.py

```diff
@@ -0,0 +1,76 @@
+class BinaryHeap:
+    
+    def __init__(self, min=True):
+        self.arr = []
+        self.size = 0
+        self._sign = 2*min - 1
```

Comment:
This is an interesting approach. Another thing you can do is keep track of a comparator function `less_than(a,b)`. If you use that to compare heap values throughout you can turn min heap into max heap by defining `less_than(a,b) = lambda x,y : x > y`

This also lets you solve the one with triples without adding the additional distance figure. You can just initialize the heap with `less_than(x,y) = lambda x,y: x[0]**2 + x[1]**2 + x[2]**2 < y[0]**2 + y[1]**2 + y[2]**2`. Not necessarily better, just a different style 🤷‍♂️